onnxruntime/core/providers/openvino/qdq_transformations/qdq_stripping.cc

```diff
@@ -0,0 +1,671 @@
+}
+
+// Current Target -> Q -> DQ -> Next Target
+// Do the inverse of the function above. Check to keep the Q if the next target is valid
+static bool IsNextTargetNodeOfQValid(const Node* Q,
+                                     const Node* current_target,
+                                     const onnxruntime::GraphViewer& src_graph,
+                                     const std::vector<std::string>& supported_ops,
+                                     bool check_const_init) {
+  const Node* next_target = GetFirstComputeOpBelowThisQ(Q);
+
+  if (std::find(supported_ops.begin(), supported_ops.end(), next_target->OpType()) != supported_ops.end()) {
+    // Always check const inits if Add is the next target
+    if ((check_const_init || next_target->OpType() == "Add") && IsAnyDQAConstantInitializer(next_target, src_graph)) {
+      return false;  // because Add/Mul/Div with const init inputs are not supported
+    } else if (next_target->OpType() == "Conv" || next_target->OpType() == "MatMul") {
+      // If any DQ of this Conv/MatMul is sandwiched between Softmax and Conv/MatMul, then don't keep it
+      bool is_valid = true;
+      for (Node::NodeConstIterator it_sw = next_target->InputNodesBegin();
+           it_sw != next_target->InputNodesEnd(); ++it_sw) {
+        const auto& sw_dq_node = &*it_sw;
+        if (sw_dq_node->OpType() != "DequantizeLinear") return true;
+        if (IsQDQSandwichedBetweenSoftmaxAndConvMatMulOps(sw_dq_node))
+          is_valid &= false;
+      }
+      return is_valid;
+    } else {
+      return true;  // because the next target is supported
+    }
+  } else if (current_target->OpType() == "Conv" || current_target->OpType() == "MatMul") {
+    return true;  // Conv and MatMul can keep all Qs by default. Is there a better way to check this?
+  } else {
+    return false;  // because the next target is not supported
+  }
+}
+
+static bool CheckDQRuleSet(const NodeUnit& node_unit,
+                           const Node* dq_node,
+                           const onnxruntime::GraphViewer& src_graph,
+                           SkipReason& reason) {
+  const auto& target_node = node_unit.GetNode();
+  auto op_type = node_unit.OpType();
+
+  // #1 Reverse DQ duplication
+  if (dq_node->Name().find("/duplicated") != std::string::npos) {
+    reason = SkipReason::DuplicateDQ;
+    return false;
+  }
+
+  // #2 If it's a constant initializer feeding to even unsupported ops with an unsupported type, keep it
+  // TODO(sspintel): check if this needs to be done only for certain supported Ops
+  if (src_graph.IsConstantInitializer(dq_node->InputDefs().at(0)->Name(), true)) {
+    return true;
+  }
+
+  // #3 If UInt16 DQ, don't keep it
+  if (GetZeroPointDT(dq_node) == ONNX_NAMESPACE::TensorProto_DataType_UINT16) {
+    reason = SkipReason::Uint16QDQ;
+    return false;
+  }
+
+  // DQs in Double QDQ cases should be kept
+  if (dq_node->InputDefs().at(2)->Name().find("zero_point_convert") != std::string::npos &&
```

Comment:
need to check the optional input exists

Reply:
Now, checking for scale_convert instead of zero_point_convert.